capstone-backend/src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -0,0 +1,181 @@
+      e.printStackTrace();
+    }
+    
+    response.setContentType("application/json;");
+    Gson gson = new GsonBuilder().disableHtmlEscaping().create();
+
+    response.getWriter().println(gson.toJson(convertResponseToVolumeData(fullOutput)));
+  }
+
+  /*
+   * Takes in a response from the Google Books API and converts it to a Collection of VolumeData objects.
+   * @param output the response from the Books API
+   * @return a Collection of the books from the API call, in VolumeData form
+   */
+  @VisibleForTesting
+  static Collection<VolumeData> convertResponseToVolumeData(String output) {
+    ArrayList<VolumeData> volumes = new ArrayList<>();
+
+    // Parse data into json object
+    JsonParser parser = new JsonParser();
+    JsonElement data = parser.parse(output);
+    JsonObject dataInfo = data.getAsJsonObject();
+
+    // Get the items element from the output and convert it into the array of book jsons
+    JsonElement items = dataInfo.get("items");
+    JsonArray itemsInfo = items.getAsJsonArray();
+
+    for (JsonElement book: itemsInfo) {
+      JsonObject bookInfo = book.getAsJsonObject();
+      String id = bookInfo.get("id").getAsString();
+
+      // volumeInfo houses all of the information about the book itself: description, authors, title
+      JsonElement volumeInfo = bookInfo.get("volumeInfo");
+      if (volumeInfo == null) continue;
+      JsonObject volumeInfoObj = volumeInfo.getAsJsonObject();
+
+      JsonElement titleElement = volumeInfoObj.get("title");
+      String title = titleElement != null ? titleElement.getAsString() : "";
+
+      // there may be any number of authors, not guaranteed to only have 1
+      JsonElement authors = volumeInfoObj.get("authors");
+      ArrayList<String> authorNames = new ArrayList<>();
+      if (authors != null) {
+        JsonArray authorsArray = authors.getAsJsonArray();
+        for (JsonElement name : authorsArray) {
+          authorNames.add(name.getAsString());
+        }
+      }
+
+      JsonElement descElement = volumeInfoObj.get("description");
+      String description = descElement != null ? descElement.getAsString() : "";
+
+      JsonElement thumbnailElement = volumeInfoObj.get("imageLinks");
+      String thumbnailLink = thumbnailElement != null ? thumbnailElement
+                                                        .getAsJsonObject()
+                                                        .get("thumbnail")
+                                                        .getAsString()
+                                                        : "";
+      thumbnailLink = thumbnailLink.replace("http", "https");
+
+      volumes.add(new VolumeData(id, title, authorNames.toArray(new String[0]), description, thumbnailLink));
+    }
+    return volumes;
+  }
+
+  /*
+   * Adds headers to a HttpServletResponse object to enable it to be accessed by the frontend resources.
+   */
+  public static HttpServletResponse addHeadersToResponse(HttpServletResponse response) {
```

Comment:
This should probably be removed from PR since it won't be used in production, even though you took out the part where you actually call it and the links would be different for all of us.

Reply:
True, since it's not the same I might as well take it out!